### DISCUSSION TITLE
How to use json5-Files?

### DISCUSSION THREAD

**Discussion by wol81** [intlify/vue-i18n#654] [Q&A]
According to https://vue-i18n.intlify.dev/guide/advanced/sfc.html#json5 it should be possible to use json5 as format for the i18n files.

I use
- vue 3.2.6
- vue-i18n 9.1.7
- @intlify/vue-i18n-loader 3.0.0
- typescript 4.1.6

In main.ts I do
`import de from "./i18n/de.json";
import en from "./i18n/en.json";`

I added the following section to my vue.config.js chainWebpack
`        config.module
            .rule('i18n')
            .test(/\.json$/)
            .exclude
                .clear()
            .end()
            .include
                .add(function() {
                    return [
                        './src/i18n'
                    ];
                })
            .end()        
            .type("javascript/auto")
            .use('@intlify/vue-i18n-loader')
            .loader('@intlify/vue-i18n-loader');`

In my de.json I added a test string
        adminCaption : "Administration"

and it complains about wrong json syntax.  

I also put the i18n in a file "de.json5", and adapted the above rule in vue.config.js accordingly, and typescript complained about 
`TS2307: Cannot find module './i18n/de.json5' or its corresponding type declarations.`

In my tsconfig.json, the setting 
`        "resolveJsonModule" : true,`
is set.

For the future (once I have integrated the higher versions), I would like to use the feature https://vue-i18n.intlify.dev/guide/advanced/typescript.html#type-safe-resources-with-schema

Question: Is this kind of setup possible?  Can one use json5 i18n files in a typescript project somehow?  If yes, would the "type-safe-resources-with-schema"-thing still work?

Currently, I'm not sure wether the problem is my inability to configure it correctly, or some kind of missing feature support at tool side.

Thanks for any hints.

**Comment by kazupon** [comment]
> For the future (once I have integrated the higher versions), I would like to use the feature https://vue-i18n.intlify.dev/guide/advanced/typescript.html#type-safe-resources-with-schema
> 
> Question: Is this kind of setup possible? Can one use json5 i18n files in a typescript project somehow? If yes, would the "type-safe-resources-with-schema"-thing still work?

Sorry, our poor documentation.

Unfortunately, for type-safe resource with schema, which is supported in v9.2, only `JSON` is supported. This is because TypeScript can't handle other `ya?ml` or `json5`.

We'll update docs on these limitations.
